xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -49,169 +48,207 @@
-      final List<ChannelCreds> channelCredsList;
-      if (channelCredsListFromResolvedAddresses == null) {
-        channelCredsList = Collections.emptyList();
-      } else {
-        channelCredsList = channelCredsListFromResolvedAddresses;
-      }
-
-      LoadBalancerProvider childBalancerProvider = new LoadBalancerProvider() {
-        @Override
-        public boolean isAvailable() {
-          return true;
-        }
-
-        @Override
-        public int getPriority() {
-          return 5;
-        }
-
-        /**
-         * A synthetic policy name identified by balancerName. The implementation detail doesn't
-         * matter.
-         */
-        @Override
-        public String getPolicyName() {
-          return "xds_child_policy_balancer_name_" + newBalancerName;
+        // This is the EDS-only with bootstrap usecase.
+        final BootstrapInfo bootstrapInfo;
+        try {
+          bootstrapInfo = bootstrapper.readBootstrap();
+        } catch (Exception e) {
+          lookasideLbHelper.updateBalancingState(
+              TRANSIENT_FAILURE,
+              new ErrorPicker(Status.UNAVAILABLE.withCause(e)));
+          return;
         }
+        final ManagedChannel channel = initLbChannel(
+            lookasideLbHelper, bootstrapInfo.getServerUri(),
+            bootstrapInfo.getChannelCredentials());
+        xdsClientRef = new RefCountedXdsClientObjectPool(new XdsClientFactory() {
+          @Override
+          XdsClient createXdsClient() {
+            // TODO(zdapeng): Replace XdsComms2 with XdsClientImpl.
+            return new XdsComms2(
+                channel, lookasideLbHelper, new ExponentialBackoffPolicy.Provider(),
+                GrpcUtil.STOPWATCH_SUPPLIER, bootstrapInfo.getNode());
+          }
+        });
+      }
 
-        @Override
-        public LoadBalancer newLoadBalancer(final Helper helper) {
-          return new LoadBalancer() {
-            @Nullable
-            XdsClient xdsClient;
-            @Nullable
-            LocalityStore localityStore;
-            @Nullable
-            LoadReportClient lrsClient;
+      // At this point the xdsClientRef is assigned in all usecases, cache them for later use.
+      this.xdsClientRef = xdsClientRef;
+      xdsClient = xdsClientRef.getObject();
+    }
 
-            @Override
-            public void handleNameResolutionError(Status error) {}
+    // TODO(zdapeng): If lrsServerName in XdsConfig is changed, call xdsClient.reportClientStats()
+    //     and/or xdsClient.cancelClientStatsReport().
+    // Note: balancerName change is unsupported and ignored.
+    // TODO(zdapeng): Remove support for balancerName.
+    // Note: childPolicy change will be handled in LocalityStore, to be implemented.
+
+    // If edsServiceName in XdsConfig is changed, do a graceful switch.
+    if (xdsConfig == null
+        || !Objects.equals(newXdsConfig.edsServiceName, xdsConfig.edsServiceName)) {
```

Comment:
Isn't `edsServiceName` just a String? But whatever, still works. Just curious.

Reply:
`Objects.equals` takes care of `null` values as well ...

Reply:
Added a protection for `XdsConfig.edsServiceName` being null.